aptos-move/parallel-executor/src/executor.rs

```diff
@@ -212,25 +262,31 @@ where
         versioned_data_cache: &MVHashMap<<T as Transaction>::Key, <T as Transaction>::Value>,
         scheduler: &'a Scheduler,
     ) -> SchedulerTask<'a> {
+        use MVHashMapError::*;
+        use MVHashMapOutput::*;
+
         let (idx_to_validate, incarnation) = version_to_validate;
         let read_set = last_input_output
             .read_set(idx_to_validate)
             .expect("Prior read-set must be recorded");
 
         let valid = read_set.iter().all(|r| {
             match versioned_data_cache.read(r.path(), idx_to_validate) {
-                Ok((version, _)) => r.validate_version(version),
-                Err(Some(_)) => false, // Dependency implies a validation failure.
-                Err(None) => r.validate_storage(),
+                Ok(Version(version, _)) => r.validate_version(version),
+                Ok(Resolved(value)) => r.validate_resolved(value),
+                Err(Dependency(_)) => false, // Dependency implies a validation failure.
+                Err(Unresolved(delta)) => r.validate_unresolved(delta),
+                Err(NotFound) => r.validate_storage(),
+                Err(DeltaApplicationFailure) => r.validate_delta_application_failure(),
```

Comment:
this should always be false? or it's optimization for the case that actual value is coincidentally 0?

Reply:
It is for the case the assumption does not hold and we have an overflow?

Reply:
what @sasha8 said (so we don't loop forever and instead panic later, when extracting outputs). Thinking is that panic is preferable to hanging, and it's also more easily testable (i.e. fake PE tests also test this case currently, that when there is an actual over/underflow, PE outputs materialized will also lead to overflow/underflow and we can panic there).

Reply:
please add a comment